### PR TITLE
Solana stake and unstake

### PR DESCRIPTION
**Pull request by georgeburduhos**

### CODE REVIEW

File: src/redux/wallets/reducer.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ const newBalance = (oldBalance: any, action: any) => ({
         : new BigNumber(oldBalance?.available).toFixed(),
     total: action.data.balance
         ? new BigNumber(action.data.balance.total).toFixed()
-        : new BigNumber(oldBalance?.total).toFixed()
+        : new BigNumber(oldBalance?.total).toFixed(),
+    detailed: action.data.balance ? action.data.balance.detailed || {} : oldBalance?.detailed || {}
```
Comment (NovacAndrei) [inline src/redux/wallets/reducer.ts:42]: Safer to use:
`action.data.balance?.detailed || oldBalance?.detailed || {}`

File: src/redux/wallets/state.ts [hunk 2]
```diff
@@ -62,6 +62,7 @@ export interface ITokenState {
         error: any;
         available: string;
         total: string;
+        detailed: any;
```
Comment (NovacAndrei) [inline src/redux/wallets/state.ts:65]: Is it possible to add a type here?

File: src/screens/pos-actions/pos-basic-action/pos-basic-action.tsx [hunk 3]
```diff
@@ -131,11 +131,12 @@ export class PosBasicActionComponent extends React.Component<
 
         const tokenConfig = getTokenConfig(this.props.blockchain, this.props.token.symbol);
 
+        const balance = new BigNumber(this.props.validators[0].amountDelegated.active).plus(
```
Comment (NovacAndrei) [inline src/screens/pos-actions/pos-basic-action/pos-basic-action.tsx:134]: Is this safe for all blockchains?

File: src/redux/wallets/reducer.ts (hunk 1 above)
Comment (NovacAndrei) [inline src/redux/wallets/reducer.ts:42]: Safer to use:
`action.data.balance?.detailed || oldBalance?.detailed || {}`

File: src/redux/wallets/state.ts (hunk 2 above)
Comment (NovacAndrei) [inline src/redux/wallets/state.ts:65]: Is is possible to add here a type?

Check `export interface IBlockchainTransaction<IAdditionalInfoType = any> {}`, maybe we can use something like this. To have easier tracking of detailed field.

**Review by NovacAndrei — APPROVED** [review]

File: src/redux/wallets/reducer.ts (hunk 1 above)
Comment (georgeburduhos) [reply src/redux/wallets/reducer.ts:42]: Fixed :)

File: src/screens/pos-actions/pos-basic-action/pos-basic-action.tsx (hunk 3 above)
Comment (georgeburduhos) [reply src/screens/pos-actions/pos-basic-action/pos-basic-action.tsx:134]: yes, pending is mandatory field added when i implemented Celo, and for near and zil its returned 0

File: src/redux/wallets/state.ts (hunk 2 above)
Comment (krisboit) [reply src/redux/wallets/state.ts:65]: It would be nice, but a bit hard in this context... it could just complicate things :)

**Review by krisboit — APPROVED** [review]

**Pull request merged**